subsys/bluetooth/host/hci_core.c

```diff
@@ -810,7 +810,7 @@ static void le_enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	}
 #endif
 
-	if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
+	if (IS_ENABLED(CONFIG_BT_PERIPHERAL)) {
```

Comment:
This has to remain CONFIG_BT_CENTRAL, as auto connect state in central role needs the scanner started again.